File: VB/CustomCommand/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit.Services
Imports DevExpress.XtraRichEdit

Namespace CustomCommand

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            ReplaceRichEditCommandFactoryService(richEditControl1)
        End Sub

        Private Sub ReplaceRichEditCommandFactoryService(ByVal control As RichEditControl)
            Dim service As IRichEditCommandFactoryService = control.GetService(Of IRichEditCommandFactoryService)()
            If service Is Nothing Then Return
            control.RemoveService(GetType(IRichEditCommandFactoryService))
            control.AddService(GetType(IRichEditCommandFactoryService), New CustomRichEditCommandFactoryService(control, service))
        End Sub
    End Class
End Namespace

File: CS/CustomCommand/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraRichEdit.Services;
using DevExpress.XtraRichEdit;

namespace CustomCommand
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            ReplaceRichEditCommandFactoryService(richEditControl1);
        }


        void ReplaceRichEditCommandFactoryService(RichEditControl control)
        {
            IRichEditCommandFactoryService service = control.GetService<IRichEditCommandFactoryService>();
            if (service == null)
                return;
            control.RemoveService(typeof(IRichEditCommandFactoryService));
            control.AddService(typeof(IRichEditCommandFactoryService), new CustomRichEditCommandFactoryService(control, service));
        }

    }
}